packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue

```diff
@@ -0,0 +1,256 @@
+      <oc-icon
+        v-else
+        name="layout-grid"
+        size="xxlarge"
+        class="space-default-image oc-px-m oc-py-m"
+      />
+    </div>
+    <div v-if="hasPeopleShares || hasLinkShares" class="oc-flex oc-flex-middle oc-mb-m">
+      <oc-icon v-if="hasPeopleShares" name="group" class="oc-mr-s" />
+      <oc-icon v-if="hasLinkShares" name="link" class="oc-mr-s" />
+      <span class="oc-text-small" v-text="shareLabel" />
+    </div>
+    <div>
+      <table class="details-table" :aria-label="detailsTableLabel">
+        <tr>
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Last activity')" />
+          <td v-text="lastModifyDate" />
+        </tr>
+        <tr v-if="space.description">
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Subtitle')" />
+          <td v-text="space.description" />
+        </tr>
+        <tr>
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Manager')" />
+          <td>
+            <span v-if="!loadOwnersTask.isRunning" v-text="ownerUsernames" />
+          </td>
+        </tr>
+        <tr>
+          <th scope="col" class="oc-pr-s" v-text="$gettext('Quota')" />
+          <td>
+            <p class="oc-mb-s oc-mt-rm" v-text="personalStorageDetailsLabel" />
+            <oc-progress
+              :value="parseInt(quotaUsagePercent)"
+              :max="100"
+              size="small"
+              :variation="quotaProgressVariant"
+            />
+          </td>
+        </tr>
+      </table>
+    </div>
+  </div>
+</template>
+<script>
+import { ref } from '@vue/composition-api'
+import Mixins from '../../../mixins'
+import MixinResources from '../../../mixins/resources'
+import { mapGetters } from 'vuex'
+import { useTask } from 'vue-concurrency'
+import { buildWebDavSpacesPath } from '../../../helpers/resources'
+import filesize from 'filesize'
+import { useStore } from 'web-pkg/src/composables'
+import { clientService } from 'web-pkg/src/services'
+
+export default {
+  name: 'SpaceDetails',
+  mixins: [Mixins, MixinResources],
+  inject: ['displayedItem'],
+  title: ($gettext) => {
+    return $gettext('Details')
+  },
+  setup() {
+    const store = useStore()
+    const spaceImage = ref('')
+    const owners = ref([])
+    const graphClient = clientService.graphAuthenticated(
+      store.getters.configuration.server,
+      store.getters.getToken
+    )
+
+    const loadImageTask = useTask(function* (signal, ref) {
+      if (!ref.space?.spaceImageData) {
+        return
+      }
+
+      const fileContents = yield ref.$client.files.getFileContents(
+        buildWebDavSpacesPath(ref.space.id, ref.space.spaceImageData.name),
+        { responseType: 'arrayBuffer' }
+      )
+
+      spaceImage.value = Buffer.from(fileContents).toString('base64')
+    })
+
+    const loadOwnersTask = useTask(function* (signal, ref) {
+      const promises = []
+      for (const userId of ref.ownerUserIds) {
+        promises.push(graphClient.users.getUser(userId))
```

Comment:
We might call listUsers in the API and call and filter the users, to do only one query